docs/01-app/01-getting-started/07-partial-prerendering.mdx

```diff
@@ -0,0 +1,227 @@
+---
+title: How to use Partial Prerendering
+nav_title: Partial Prerendering
+description: Learn how to combine the benefits of static and dynamic rendering with Partial Prerendering.
+version: experimental
+related:
+  title: Next Steps
+  description: Learn more about the config option for Partial Prerendering.
+  links:
+    - app/api-reference/config/next-config-js/ppr
+---
+
+Partial Prerendering (PPR) is a rendering strategy that allows you to combine static and dynamic content in the same route. This improves the initial page performance while still supporting personalized, dynamic data.
+
+<Image
+  alt="Partially Prerendered Product Page showing static nav and product information, and dynamic cart and recommended products"
+  srcLight="/learn/light/thinking-in-ppr.png"
+  srcDark="/learn/dark/thinking-in-ppr.png"
+  width="1600"
+  height="632"
+/>
+
+When a user visits a route:
+
+- The server sends a **shell** containing the static content, ensuring a fast initial load.
+- The shell leaves **holes** for the dynamic content that will load in asynchronously.
+- The dynamic holes are **streamed in parallel**, reducing the overall load time of the page.
+
+> **🎥 Watch:** Why PPR and how it works → [YouTube (10 minutes)](https://www.youtube.com/watch?v=MTcPrTIBkpA).
+
+## How does Partial Prerendering work?
+
+Partial Prerendering uses React's [Suspense](https://react.dev/reference/react/Suspense) to defer rendering parts of your application until some condition is met.
+
+The Suspense fallback is embedded into the initial HTML file along with the static content. At build time (or during revalidation), the static content is **prerendered** to create a static shell. The rendering of dynamic content is **postponed** until the user requests the route.
+
+Wrapping a component in Suspense doesn't make the component itself dynamic, but rather Suspense is used as a boundary between your static and dynamic code. As long as you're using Suspense, Next.js will know which parts of your route are static and which are dynamic.
```

Comment:
Let's see if I got the mental model right.

Wrapping with Suspense doesn't necessarily mean that the pre-rendering/shell-generation, step will stop here, and the fallback. 

Rather, when the pre-render step encounters something dynamic under this Suspense boundary, it will unwind back it, as a kind of checkpoint, and render the fallback instead.

Is that somewhat correct?

Reply:
This was helpful for my understanding: https://github.com/vercel/next.js/blob/7e9b55da8b0d72c22708ae08a71581858e802bef/errors/ppr-caught-error.mdx

In PPR/DynamicIO, Dynamic API usage throws an error. Suspense is a way to encapsulate the error. 

That's why, if wrap suspense around something that doesn't throw, Next.js might still treat the content as static.